mypyc/codegen/emitwrapper.py

```diff
@@ -269,6 +269,70 @@ def generate_dunder_wrapper(cl: ClassIR, fn: FuncIR, emitter: Emitter) -> str:
     return name
 
 
+def generate_setattr_wrapper(cl: ClassIR, fn: FuncIR, emitter: Emitter) -> str:
+    """Generates a wrapper for native __setattr__ methods to be able to fit into
+    the tp_setattro protocol slot. This specifically means that the function
+    raises an exception and returns -1 on failure, 0 on success
+    """
+    input_args = ', '.join('PyObject *obj_{}'.format(arg.name) for arg in fn.args)
```

Comment:
The third argument may be `NULL` for `__delattr__`. Check for this and raise an exception. Add a test for this. (No need to actually support `__delattr__` in this PR.)

If you want to support `__delattr__`, you can generate a separate wrapper method here from `__delattr__` if the class supports it. Then in the `__setattr__` wrapper you'll actually call the `__delattr__` wrapper if the third argument is `NULL`. Otherwise, you'll proceed as normal.